keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,767 @@
+- Enable Windows systems to support CPU Manager or memory manager and hence Topology aware
+  Scheduling in general.
+
+## Proposal
+
+Kube-scheduler plugin will be moved from kuberntes-sigs/scheduler-plugin (or out-of-tree)
+plugin into the main tree as a built-in plugin. This plugin implements a simplified version
+of Topology Manager and hence is different from original topology manager algorithm. Plugin
+would be disabled by default and when enabled would check for the ability to run pod only in
+case of single-numa-node policy Topology Manager policy on the node, since it is the most strict
+policy, it implies that the launch on the node with other existing policies will be successful
+if the condition for single-numa-node policy passed for the worker node.
+
+To work, this plugin requires topology information of the available resources for each NUMA cell on worker nodes.
+
+Kubelet will be responsible for collecting all necessary resource information of the pods,
+based on allocatable resources on the node and allocated resources to pods. The NUMA nodes
+would be represented as NUMA cells in Kubelet and the NodeResourceTopology would capture the 
+resource information at a NUMA cell level granularity.
+
+Once the information is captured in the NodeResourceTopology API, the scheduler can refer to
+it like it refers to Node Capacity and Allocatable while making a Topology-aware Scheduling decision.
+
+### User Stories
+
+As a Kubernetes cluster operator managing a cluster with multiple bare metal worker nodes with NUMA
+topology and Topology Manager enabled, I want the scheduler to be Topology-aware in order to ensure that the
+pods are only placed on nodes where requested resources can be appropriately aligned by Topology Manager based on its policy.
+The scheduler shouldn't send the pod to the node where kubelet will reject it with "Topology Affinity Error".
+This issue leads to runaway pod creation if the pod is part of a Deployment or ReplicaSet as the associated controllers
+notices the pod failure and keep creating another pod.
+
+
+### Risks and Mitigations
+
+Topology Manager on the worker node knows exact resources and their NUMA node allocated to pods but the and node resource
+topology information is delivered to Topology aware scheduler plugin with latency meaning that the scheduler will not know
+actual NUMA topology until the information of the available resources at a NUMA node level is evaluated in the kubelet which
+could still lead to scheduling of pods to nodes where they won't be admitted by Topology Manager.
+
+This can be mitigated if kube-scheduler provides a hint of which NUMA ID a pod should be assigned and Topology Manager on the
+worker node takes that into account.
+
+## Design Details
+
+- add a new flag in Kubelet called `ExposeNodeResourceTopology` in the kubelet config or command line argument called `expose-noderesourcetopology` which allows 
```

Comment:
Take a decision here: Are you proposing a permanent command line flag or not? Note that this is different from the feature gate, which should disappear in a a few releases